TYPO3.Flow/Migrations/Code/Version20151113161300.php

```diff
@@ -0,0 +1,161 @@
+<?php
+namespace TYPO3\Flow\Core\Migrations;
+
+/*
+ * This file is part of the TYPO3.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\Flow\Configuration\ConfigurationManager;
+use TYPO3\Flow\Security\RequestPattern\ControllerObjectName;
+use TYPO3\Flow\Security\RequestPattern\CsrfProtection;
+use TYPO3\Flow\Security\RequestPattern\Host;
+use TYPO3\Flow\Security\RequestPattern\Ip;
+use TYPO3\Flow\Security\RequestPattern\Uri;
+
+/**
+ * Adjust "Settings.yaml" to new "requestPattern" and "firewall" syntax (see FLOW-412)
+ */
+class Version20151113161300 extends AbstractMigration
```

Comment:
Right, this is done by the migration and only for 3rd party packages that used either of the patterns or filters..

Good point, I wasn't sure about this one..
But unfortunately it's a bit tricky.

So for the filters:
Previously we used the array syntax (rather than a dictionary) leading to numeric keys.
So if someone specified one filter for example it would override the first (and only) defined filter of the Flow package, being the `CsrfProtection` (!).

I guess it makes sense to change it to `Package.Key:<Filter>`, so `TYPO3.Flow:CsrfProtection` in this case.
I doubt that a lot of people used this while relying on the fact that a filter overrides the filters with the same index..


For the request patterns it's a bit different:

It's quite likely that a 3rd party package *overrides* the pattern(options) for an existing auth provider.. In this case if we change the name to something other than the pattern classname we risk that the final configuration is sth like:

```
requestPatterns:
  'TYPO3.Neos:ControllerObjectName':
    pattern: 'ControllerObjectName'
    patternOptions:
      controllerObjectNamePattern: 'TYPO3\Neos\Controller\.*|TYPO3\Neos\Service\.*|TYPO3\Media\Controller\.*'
  'My.Package:ControllerObjectName':
    pattern: 'ControllerObjectName'
    patternOptions:
      controllerObjectNamePattern: 'Overridden...'

ending in a *merge* rather than the original *override*